hibernate-infinispan/src/main/java/org/hibernate/cache/infinispan/util/Caches.java

```diff
@@ -291,6 +291,10 @@ public static void removeAll(AdvancedCache cache) {
 	}
 
 	public static CollectableCloseableIterable keys(AdvancedCache cache) {
+		if (cache.getCacheConfiguration().transaction().transactionMode().isTransactional()) {
+			// Dummy read to enlist the LocalTransaction as workaround for ISPN-5676
+			cache.containsKey(false);
```

Comment:
I think that in all places where I use this, local cache is passed to the method. I don't think that the additional flag should be passed at this place (which is some kind of generic helper on Cache API).